File: solution/0000-0099/0006.Zigzag Conversion/Solution.java
class Solution {
    public String convert(String s, int numRows) {
        if (numRows == 1) {
            return s;
        }
        StringBuilder[] lines = new StringBuilder[numRows];
        Arrays.setAll(lines, l -> new StringBuilder());
        int i = 0, step = 1;
        for (char c : s.toCharArray()) {
            lines[i].append(c);
            i += step;
            if (i == 0 || i == numRows-1) {
                step = -step;
            }
        }
        return String.join("", lines);
    }
}

File: solution/0000-0099/0006.Zigzag Conversion/Solution.go
func convert(s string, numRows int) string {
    if numRows == 1 {
        return s
    }

    lines := make([][]byte, numRows)
    i, step := 0, 1
    for _, c := range s {
        lines[i] = append(lines[i], byte(c))
        i += step
        if i == 0 || i == numRows-1 {
            step = -step;
        }
    }
    return string(bytes.Join(lines, nil))
}
